integration/tests/dry_run/dry_run_test.go

```diff
@@ -216,7 +223,59 @@ var _ = Describe("(Integration) [Dry-Run test]", func() {
 		Entry("set profile option", "--profile=aws"),
 	)
 
-	Describe("create cluster and nodegroups from the output of dry-run", func() {
+	DescribeTable("create cluster with instance selector options", func(setValues func(actual, expected *api.ClusterConfig), createArgs ...string) {
+		cmd := params.EksctlCreateCmd.
+			WithArgs(
+				"cluster",
+				"--name=dry-run-cluster",
+				"--dry-run",
+				"--nodegroup-name=ng-default",
+			).
+			WithArgs(createArgs...)
+
+		session := cmd.Run()
+		Expect(session.ExitCode()).To(Equal(0))
+		output := session.Buffer().Contents()
+
+		actual, expected := parseOutput(output)
+		setValues(actual, expected)
+		Expect(actual).To(Equal(expected))
+
+	}, Entry("instance selector options with unmanaged nodegroup", func(actual, expected *api.ClusterConfig) {
+		// This does not do an exact match because instance types matching the instance selector criteria may
+		// change over time as EC2 adds more instance types
+		Expect(actual.NodeGroups[0].InstancesDistribution.InstanceTypes).ToNot(BeEmpty())
+		actual.NodeGroups[0].InstancesDistribution.InstanceTypes = nil
+
+		expected.ManagedNodeGroups = nil
+		ng := expected.NodeGroups[0]
+		ng.InstancesDistribution = &api.NodeGroupInstancesDistribution{
+			InstanceTypes: nil,
+		}
+		ng.InstanceType = "mixed"
+		ng.InstanceSelector = &api.InstanceSelector{
+			VCPUs:  2,
+			Memory: "4",
+		}
+
+	}, "--instance-selector-vcpus=2", "--instance-selector-memory=4"),
+
+		Entry("instance selector options with managed nodegroup", func(actual, expected *api.ClusterConfig) {
+			Expect(actual.ManagedNodeGroups[0].InstanceTypes).ToNot(BeEmpty())
+			actual.ManagedNodeGroups[0].InstanceTypes = nil
+
+			expected.NodeGroups = nil
+			ng := expected.ManagedNodeGroups[0]
+			ng.InstanceTypes = nil
+			ng.InstanceSelector = &api.InstanceSelector{
+				VCPUs:           2,
+				Memory:          "4",
+				CPUArchitecture: "x86_64",
+			}
+		}, "--managed", "--instance-selector-vcpus=2", "--instance-selector-memory=4", "--instance-selector-cpu-architecture=x86_64"),
+	)
+
+	PDescribe("create cluster and nodegroups from the output of dry-run", func() {
```

Comment:
should this be Pending?

Reply:
Good catch, I forgot to update it. Thanks!